src/endpoints/assets/v1.ts

```diff
@@ -0,0 +1,618 @@
+import type {APIClient} from '@wharfkit/antelope'
+import type {Int32Type, NameType, UInt32Type, UInt64Type} from '@wharfkit/antelope'
+import type {ActionType} from '../../contracts/atomicassets'
+import {Assets} from '../../types'
+import type {OfferState} from '../../types'
+
+export class AssetsV1APIClient {
+    constructor(private client: APIClient) {}
+
+    async get_assets(
+        options?: {
+            collection_name?: NameType[]
+            schema_name?: NameType[]
+            template_id?: Int32Type[]
+            burned?: boolean
+            owner?: NameType[]
+            match?: string
+            search?: string
+            match_immutable_name?: string
+            match_mutable_name?: string
+            is_transferable?: boolean
+            is_burnable?: boolean
+            minter?: NameType[]
+            burner?: NameType[]
+            initial_receiver?: NameType[]
+            collection_blacklist?: NameType[]
+            collection_whitelist?: NameType[]
+            only_duplicate_templates?: boolean
+            has_backed_tokens?: boolean
+            authorized_account?: NameType
+            template_whitelist?: Int32Type[]
+            template_blacklist?: Int32Type[]
+            hide_templates_by_accounts?: NameType[]
+            hide_offers?: boolean
+            asset_id?: UInt64Type[]
+            ids?: UInt64Type[]
+            lower_bound?: string
+            upper_bound?: string
+            before?: number
+            after?: number
+            page?: number
+            limit?: number
+            order?: 'asc' | 'desc'
+            sort?: 'asset_id' | 'minted' | 'updated' | 'transferred' | 'template_mint' | 'name'
+        },
+        extra_options?: {[key: string]: string}
+    ) {
+        const queryParts = {}
+
+        for (const [key, value] of Object.entries(options || {})) {
+            queryParts[key] = value as string
```

Comment:
Should we be casting as string here? Some of these options should be treated as strings (eg. `NameType` which could be a `Name`)

Reply:
`as string` is a Typescript compile time thing so it doesn't do any at runtime